go/fn/object.go

```diff
@@ -169,20 +169,41 @@ func (o *SubObject) NestedSliceOrDie(fields ...string) SliceSubObjects {
 	return val
 }
 
-// NestedInt64 returns the int64 value, if the field exist and a potential error.
-func (o *SubObject) NestedMap(fields ...string) (*SubObject, bool, error) {
-	var variant internal.MapVariant
+// NestedMap returns a map[string]string value of a nested field, false if not found and an error if not a map[string]string type.
+func (o *SubObject) NestedStringMap(fields ...string) (map[string]string, bool, error) {
+	var variant map[string]string
 	found, err := o.Get(&variant, fields...)
 	if err != nil || !found {
 		return nil, found, err
 	}
-	return &SubObject{obj: &variant}, found, err
+	return variant, found, err
 }
 
-// NestedInt64OrDie returns the string value at fields. An empty string will be
-// returned if the field is not found. It will panic if encountering any errors.
-func (o *SubObject) NestedMapOrDie(fields ...string) *SubObject {
-	val, _, err := o.NestedMap(fields...)
+// NestedMapOrDie returns a map[string]string value of a nested field.
```

Comment:
godoc out-of-date.
Please audit the godoc. It seems many are out-of-date.

Reply:
Same as https://github.com/GoogleContainerTools/kpt-functions-sdk/pull/573#discussion_r875107750, will do that in a follow up PR since these two have conflicts.